### PR TITLE
Major Refactor

### PR DESCRIPTION
**Pull request by icholy**
This is a pretty big refactor that introduce breaking changes. Let me know what you think.

## Breaking changes
* Add `Date` & `Time` types.
* Convert numeric fields to either `float64` or `int64`.
* Rename `Sentence` to `Sent`
* Rename `Sent.GetSentence()` to `Sent.Sentence()`
* Rename `SentenceI` to `Message`
* Rename `NewLatLong` to `ParseLatLong`

## Non-Breaking changes
* Add `ParseSentence(string)`
* Add `Sent.Prefix() string`
* Add `nmea: ` prefix to error messages.
* Unify error message format.
* Add `parser` to simplify sentence parsing.

### CODE REVIEW

File: parser.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	}
+	return p
+}
+
+// Err returns the first error encounterd during the parser's usage.
+func (p *parser) Err() error {
+	return p.err
+}
+
+// SetErr assigns the an error. Calling this method has no
```
Comment (bogdanprodanj) [inline parser.go]: Fix typo

Comment (icholy) [reply parser.go]: @bogdanprodanj I doubt this is going to get merged.

**Comment by icholy** [comment]
@adrianmo I didn't notice your message till right now. I'll update the pr asap.

**Comment by adrianmo** [comment]
Great @icholy :)

**Comment by icholy** [comment]
@adrianmo ready for review. I skipped the test for `Sentence.Validate`, I don't think it makes sense to test a function that returns `nil`. But if you really want me to, let me know and I'll add it.

**Comment by adrianmo** [comment]
@icholy thank you for making the changes. I'm merging this PR now.

**Pull request merged**